src/Nri/Ui/Tooltip/V2.elm

```diff
@@ -78,37 +81,47 @@ import Svg.Styled as Svg exposing (Svg, circle, g, svg)
 import Svg.Styled.Attributes exposing (cx, cy, d, fill, fillRule, height, r, stroke, strokeWidth, viewBox, width)
 
 
-{-| A standard NoRedInk tooltip, which appears around its parent element.
+{-| -}
+type Attribute msg
+    = Attribute (Tooltip msg -> Tooltip msg)
```

Comment:
Mostly I like that it's flexible & easy to get started with an unfamiliar component. 

The list-based approach with defaults makes it easy to get your starting point version of the component up & visible. And the list makes it really easy to slot in new attributes as-needed.

I also think the list-based-attributes approach is familiar from the Html module. And using the same patterns across our component library should make picking up an unfamiliar component easier.

What drawbacks are you concerned about with the list-of-attributes API?

Reply:
oops, I should have been more specific. The list-of-attributes API is great. I even recommended it in a talk once 😆 

I meant, why is it:

```elm
type Options = …

type Attribute = Attribute (Options -> Options)
```

instead of

```elm
type Options = ...

type Attribute = Some | Concrete | Things

customize : Attribute -> Options -> Options

view : List Attribute -> Html msg
view attributes =
    let final = List.foldl customize defaultOptions attributes
    in ...
```